src/cmd/cluster_intel_collector/config.go

```diff
@@ -0,0 +1,90 @@
+package cluster_intel_collector
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	log = newLogCollectorLogger("[log collector]")
+)
+
+type intelCollectorContext struct {
```

Comment:
We have different kind of collectors, packing them all into one type isn't good software design. The context provides access to general stuff all of the collectors need. Otherwise the collectors shall be kept separated. The context itself fullfills no collection tasks whatever.